components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/resources/store-api.yaml

```diff
@@ -234,6 +234,62 @@ paths:
           schema:
             $ref: '#/definitions/Error'
 
+
+  /apis/{apiId}/generate-sdk/{language}:
```

Comment:
Sorry for telling this bit late but have we discussed this before to add the language as a path parameter? This is a control-resource basically we should try to provide arguments as query parameters.

Options are:

/apis/{apiId}/generate-sdk?language={}
/apis/generate-sdk?apiId={id}&language={lang}

Reply:
The reason to use "apiId" and "language" as path parameters is because they are not optional. Those two should be there to generate an SDK.

Reply:
For control-resources query parameters does not need to be optional. We can mandate that. That is a special case.